src/webhook/webhooks.ts

```diff
@@ -0,0 +1,17 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import { receiverOn } from "./on";
+import { receiverHandle } from "./receive";
+import { State, WebhookContext } from "./types";
+
+export class Webhooks {
+	public on: (event: string | string[], callback: Function) => void;
+	public receive: (eventPayload: WebhookContext) => Promise<void[]>;
+
+	constructor() {
+		const state: State = {
+			hooks: {}
+		};
+		this.on = receiverOn.bind(null, state);
+		this.receive = receiverHandle.bind(null, state);
+	}
+}
```

Comment:
Not sure I understand the point of this class

Reply:
This class holds the information about registered events and their handler.  For e.g . to register for push events
webhooks.on("push", (event: WebhookContext) => {... }); 

The receive method is invoked when any incoming webhook event is received. This method decides based on event and action, which event handlers to be executed.

Reply:
Maybe some other name (e.g. WebhooksRouter) would make it more clear? (Although, Router is kinda express thingy... maybe Orchestrator or Relay...)